### DISCUSSION TITLE
Query about the plot API

### DISCUSSION THREAD

**Discussion by forsakendaemon** [arviz-devs/arviz#1770] [Q&A]
It feels a little strange to me that when using the matplotlib backend, Arviz returns the array of axes rather than returning a Figure directly - when wanting to use the return value of a `plot_*` call, I often have to instantiate an appropriately-configured Figure after the fact, which replicates work that Arviz has already done and is annoyingly finicky to get right.

Chasing through the code I can see that keeping track of the Figure might require a fair amount of care, and it would be a breaking API change to just change it wholesale, so I was wondering whether there was a reason for that choice that I might not be able to see at the moment - I have the feeling that there's something that I'm missing, I'm just not sure what it is.

**Comment by OriolAbril** [comment]
I don't really know because I wasn't here yet, but I had the feeling it was because it's easier to add more plots and customize the appearance having the axes, but there also is https://github.com/arviz-devs/arviz/issues/276 which suggests an alternative to this approach. It's something that always kind of made sense to me but I can't really say why.

As a quick workaround, you can get the figure from an axes using `axes.get_figure()` which we actually use in https://github.com/arviz-devs/arviz/blob/main/arviz/plots/backends/matplotlib/elpdplot.py#L82 so you can also get the figure from the axes returned by arviz functions.

**Reply by forsakendaemon** [reply]
I totally missed `axes.get_figure()`!

That obviates several hours of hacking... Thanks for the heads up! :)
